File: Number Generator/Python/generator.py
import random
import time

LOWER_LIMIT = 1
UPPER_LIMIT = 2147483647

def generate_and_save_random_numbers(file_path, num_numbers):
    with open(file_path, 'w') as file:
        for _ in range(num_numbers):
            random_number = random.randint(LOWER_LIMIT, UPPER_LIMIT)
            file.write(str(random_number) + '\n')

def write_and_print_data(file, numbers_generated, execution_time):
    print(f"Generated {numbers_generated} numbers, Execution time: {execution_time} seconds")
    file.write(f"{numbers_generated},{execution_time}\n")

def generate(file_path, numbers_generated, log_file):
    start_time = time.time()
    generate_and_save_random_numbers(file_path, numbers_generated)
    end_time = time.time()
    execution_time = end_time - start_time
    write_and_print_data(log_file, numbers_generated, execution_time)


with open("number_generator_data.csv", 'w') as log_file:
    log_file.write("Numbers,Time\n")

    for i in range(1, 9):
        numbers_generated = 10 ** i
        file_path = f'numbers{numbers_generated}.txt'
        generate(file_path, numbers_generated, log_file)

File: Number Generator/C++/generator.cpp
#include <iostream>
#include <fstream>
#include <cstdlib>
#include <ctime>

const int LOWER_LIMIT = 1;
const int UPPER_LIMIT = 2147483647;

void generateAndSaveRandomNumbers(const std::string& filePath, int numNumbers) {
    std::ofstream file(filePath);
    if (!file.is_open()) {
        std::cerr << "Error opening file: " << filePath << std::endl;
        return;
    }

    for (int i = 0; i < numNumbers; ++i) {
        int randomNumber = rand() % (UPPER_LIMIT - LOWER_LIMIT + 1) + LOWER_LIMIT;
        file << randomNumber << '\n';
    }

    file.close();
}

void writeAndPrintData(std::ofstream& file, int numbersGenerated, double executionTime) {
    std::cout << "Generated " << numbersGenerated << " numbers, Execution time: " << executionTime << " seconds\n";
    file << numbersGenerated << "," << executionTime << '\n';
}

void generate(const std::string& filePath, int numbersGenerated, std::ofstream& logFile) {
    std::clock_t startTime = std::clock();

    generateAndSaveRandomNumbers(filePath, numbersGenerated);

    std::clock_t endTime = std::clock();
    double executionTime = static_cast<double>(endTime - startTime) / CLOCKS_PER_SEC;

    writeAndPrintData(logFile, numbersGenerated, executionTime);
}

int main() {
    std::ofstream logFile("../number_generator_data.csv");
    if (!logFile.is_open()) {
        std::cerr << "Error opening log file." << std::endl;
        return 1;
    }

    logFile << "Numbers,Time\n";

    for (int i = 1; i < 9; ++i) {
        int numbersGenerated = static_cast<int>(pow(10, i));
        std::string filePath = "../numbers" + std::to_string(numbersGenerated) + ".txt";
        generate(filePath, numbersGenerated, logFile);
    }

    logFile.close();

    return 0;
}
